### PR TITLE
[TIMOB-20228] iOS: Remove IS_XCODE_7 macros

### PR DESCRIPTION
**Pull request by hansemannn**
JIRA: https://jira.appcelerator.org/browse/TIMOB-20228

Please ensure, that travis does only build with Xcode 7.2 (or later) from now on to prevent CI build fails.

### CODE REVIEW

**Comment by hansemannn** [comment]
For some reason, travis keeps failing, but `scons` and Xcode builds work:
```
[ERROR] The following build commands failed:
[ERROR] 	CompileC build/Intermediates/testApp.build/Debug-iphonesimulator/testApp.build/Objects-normal/i386/TiUITextWidget.o Classes/TiUITextWidget.m normal i386 objective-c com.apple.compilers.llvm.clang.1_0.compiler
[ERROR] (1 failure)
```

**Comment by cheekiatng** [comment]
Yes this will fail for sure because the travis CI is running on Xcode 6.4. The Xcode 7.2 image on travis ci somehow doesn't sit well with our unit test code, so we have to look at using Jenkins soon. Don't worry about the PR failing, and don't merge this yet.

**Comment by hansemannn** [comment]
Alright! It will also work with Xcode 7.1, but let's just wait here.

File: iphone/Classes/MediaModule.m [hunk 1]
```diff
@@ -29,14 +29,11 @@
 #import <MobileCoreServices/UTCoreTypes.h>
 #import <QuartzCore/QuartzCore.h>
 #import <AVFoundation/AVFoundation.h>
-
 #import <UIKit/UIPopoverController.h>
-
-#if IS_XCODE_7_1
 #import <Photos/Photos.h>
 #import "TiUIiOSLivePhoto.h"
```
Comment (AngelkPetkov) [inline iphone/Classes/MediaModule.m]: The ticket is for minimum Xcode 7.0 , so should we be removing this , seeing as livePhotView is iOS 9.1 onwards not iOS 9.0.

File: iphone/Classes/MediaModule.m [hunk 2]
```diff
@@ -1899,14 +1889,11 @@ - (void)imagePickerController:(UIImagePickerController *)picker_ didFinishPickin
             }
         }
         
-#if IS_XCODE_7_1
```
Comment (AngelkPetkov) [inline iphone/Classes/MediaModule.m]: Same as above.

File: iphone/Classes/TiApp.m [hunk 3]
```diff
@@ -291,13 +289,13 @@ - (void)booted:(id)bridge
...
+
         if(launchedShortcutItem != nil) {
             [self handleShortcutItem:launchedShortcutItem waitForBootIfNotLaunched:YES];
             launchedShortcutItem = nil;
         }
-#endif
-		if (localNotification != nil) {
+
+        if (localNotification != nil) {
 			[[NSNotificationCenter defaultCenter] postNotificationName:kTiLocalNotification object:localNotification userInfo:nil];
```
Comment (AngelkPetkov) [inline iphone/Classes/TiApp.m]: Indentation

File: iphone/Classes/TiContactsGroup.m [hunk 4]
```diff
@@ -157,8 +156,8 @@ -(NSArray*)members:(id)unused
 			return nil;
 		}
 	}
-#endif
-	CFArrayRef arrayRef = ABGroupCopyArrayOfAllMembers([self record]);
+
+    CFArrayRef arrayRef = ABGroupCopyArrayOfAllMembers([self record]);
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsGroup.m]: Indentation

File: iphone/Classes/TiContactsGroup.m [hunk 5]
```diff
@@ -183,8 +182,8 @@ -(NSArray*)sortedMembers:(id)value
 		TiThreadPerformOnMainThread(^{result = [[self sortedMembers:value] retain];}, YES);
 		return [result autorelease];
 	}
-#if IS_XCODE_7
-	if ([TiUtils isIOS9OrGreater]) {
+
+    if ([TiUtils isIOS9OrGreater]) {
 		CNContactStore *ourContactStore = [module contactStore];
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsGroup.m]: Indentation for the chunk of code below

File: iphone/Classes/TiContactsGroup.m [hunk 6]
```diff
@@ -260,8 +259,8 @@ -(void)add:(id)arg
 {
 	ENSURE_SINGLE_ARG(arg,TiContactsPerson)
 	ENSURE_UI_THREAD(add,arg);
-#if IS_XCODE_7
-	if ([TiUtils isIOS9OrGreater]) {
+
+    if ([TiUtils isIOS9OrGreater]) {
 		TiContactsPerson *person = arg;
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsGroup.m]: Indentation

File: iphone/Classes/TiContactsGroup.m [hunk 7]
```diff
@@ -298,8 +297,8 @@ -(void)remove:(id)arg
 {
 	ENSURE_SINGLE_ARG(arg,TiContactsPerson)
 	ENSURE_UI_THREAD(remove,arg);
-#if IS_XCODE_7
-	if ([TiUtils isIOS9OrGreater]) {
+
+    if ([TiUtils isIOS9OrGreater]) {
 		TiContactsPerson *person = arg;
 		CNContactStore *ourContactStore = [module contactStore];
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsGroup.m]: Indentation

File: iphone/Classes/TiContactsGroup.m [hunk 8]
```diff
@@ -319,8 +318,8 @@ -(void)remove:(id)arg
 		RELEASE_TO_NIL(saveRequest)
 		return;
 	}
-#endif
-	CFErrorRef error;
+
+    CFErrorRef error;
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsGroup.m]: Indentation

File: iphone/Classes/TiContactsPerson.m [hunk 9]
```diff
@@ -605,16 +597,16 @@ -(TiBlob*)image
 -(void)setBirthday:(NSString*)date
 {
 	ENSURE_UI_THREAD(setBirthday, date)
-#if IS_XCODE_7
-	if ([TiUtils isIOS9OrGreater]) {
+
+    if ([TiUtils isIOS9OrGreater]) {
 		NSDate *saveDate = [TiUtils dateForUTCDate:date];
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsPerson.m]: Indentation , all the indentation looks good in code but messed up on github. Think some its spaces and you are using tabs or the other way around

File: iphone/Classes/TiContactsPerson.m [hunk 10]
```diff
@@ -579,16 +571,16 @@ -(TiBlob*)image
 		TiThreadPerformOnMainThread(^{result = [[self image] retain];}, YES);
 		return [result autorelease];
 	}
-#if IS_XCODE_7
-	if ([TiUtils isIOS9OrGreater]) {
+
+    if ([TiUtils isIOS9OrGreater]) {
 		if (person.imageDataAvailable == YES) {
```
Comment (AngelkPetkov) [inline iphone/Classes/TiContactsPerson.m]: Indentation

File: iphone/Classes/MediaModule.m (hunk 1 above)
Comment (hansemannn) [inline iphone/Classes/MediaModule.m]: We will set the requirements to at least Xcode 7.1 (probably 7.3 or whatever is the latest version before the 6.0.0 release).

File: iphone/Classes/TiApp.m (hunk 3 above)
Comment (hansemannn) [inline iphone/Classes/TiApp.m]: Where? I fixed the wrong indentation by doing the above

**Comment by hansemannn** [comment]
Closing because of weird merge conflicts. Will setup a PR from current code base.

**Pull request closed without merging**